build_your_own_images.md

```diff
@@ -0,0 +1,185 @@
+#     if [ "$ASSET" = "remote" ] ; then \
+#       CODENAME=$(cat /etc/os-release | grep VERSION_CODENAME | cut -d = -f 2) \
+#       && DOWNLOAD_URL="https://download.konghq.com/gateway-${KONG_VERSION%%.*}.x-debian-${CODENAME}/pool/all/k/kong/kong_${KONG_VERSION}_amd64.deb" \
+#       && curl -fL $DOWNLOAD_URL -o /tmp/kong.deb \
+#       && echo "$KONG_SHA256  /tmp/kong.deb" | sha256sum -c -; \
+#     fi \
+#     && apt-get update \
+#     && apt-get install --yes /tmp/kong.deb \
+#     && rm -rf /var/lib/apt/lists/* \
+#     && rm -rf /tmp/kong.deb \
+#     && chown kong:0 /usr/local/bin/kong \
+#     && chown -R kong:0 /usr/local/kong \
+#     && ln -s /usr/local/openresty/bin/resty /usr/local/bin/resty \
+#     && ln -s /usr/local/openresty/luajit/bin/luajit /usr/local/bin/luajit \
+#     && ln -s /usr/local/openresty/luajit/bin/luajit /usr/local/bin/lua \
+#     && ln -s /usr/local/openresty/nginx/sbin/nginx /usr/local/bin/nginx \
+#     && kong version \
+#    && apt-get purge curl -y
+
+# Uncomment the following section if you are installing a .apk.tar.gz
+# Edit the DOWNLOAD_URL line to install from a repository other than 
+# download.konghq.com
+# RUN set -ex; \
+#     apk add bash curl ca-certificates; \
+#     arch="$(apk --print-arch)"; \
+#     case "${arch}" in \
+#       x86_64) export ARCH='amd64'; KONG_SHA256=$KONG_AMD64_SHA ;; \
+#       aarch64) export ARCH='arm64'; KONG_SHA256=$KONG_ARM64_SHA ;; \
+#     esac; \
+#     if [ "$ASSET" = "remote" ] ; then \
+#       curl -fL "https://download.konghq.com/gateway-${KONG_VERSION%%.*}.x-alpine/kong-${KONG_VERSION}.${ARCH}.apk.tar.gz" -o /tmp/kong.apk.tar.gz \
+#       && echo "$KONG_SHA256  /tmp/kong.apk.tar.gz" | sha256sum -c -; \
+#     fi \
+#     && apk add --no-cache --virtual .build-deps tar gzip \
+#     && tar -C / -xzf /tmp/kong.apk.tar.gz \
+#     && apk add --no-cache libstdc++ libgcc openssl pcre perl tzdata libcap zlib zlib-dev bash \
+#     && adduser -S kong \
+#     && addgroup -S kong \
+#     && mkdir -p "/usr/local/kong" \
+#     && chown -R kong:0 /usr/local/kong \
+#     && chown kong:0 /usr/local/bin/kong \
+#     && chmod -R g=u /usr/local/kong \
+#     && rm -rf /tmp/kong.tar.gz \
+#     && ln -s /usr/local/openresty/bin/resty /usr/local/bin/resty \
+#     && ln -s /usr/local/openresty/luajit/bin/luajit /usr/local/bin/luajit \
+#     && ln -s /usr/local/openresty/luajit/bin/luajit /usr/local/bin/lua \
+#     && ln -s /usr/local/openresty/nginx/sbin/nginx /usr/local/bin/nginx \
+#     && apk del .build-deps \
+#     && kong version
+
+COPY docker-entrypoint.sh /docker-entrypoint.sh
+
+USER kong
+
+ENTRYPOINT ["/docker-entrypoint.sh"]
+
+EXPOSE 8000 8443 8001 8444 $EE_PORTS
+
+STOPSIGNAL SIGQUIT
+
+HEALTHCHECK --interval=10s --timeout=10s --retries=10 CMD kong health
+
+CMD ["kong", "docker-start"]
+```
+
+### Run the docker command
+
+Run the command `docker build --no-cache -t kong-<your-base-image>
+<path-for-built-image>` to build the docker image.
```

Comment:
For posterity what are the pros / cons of reiterating the `Dockerfile` instead of for example instructing a user to use the (slim) Dockerfile's?

```
wget -O kong.deb download.konghq.com ....
docker build --build-arg ASSET=local -f Dockerfile.deb -t mytag .
```

Reply:
I have no strong feelings / this seems fine. Could / should we add a comment to our Dockerfiles that any substantive change should be updated in <final resting place of this document>

Reply:
for posterity: @hutchic reminded me that we can't parametrize the FROM argument to allow someone to pass in their own base image because Docker doesn't like that and won't approve our images.

Reply:
To summarize various offline conversations: we aren't going to parametrize the FROM arg because that slows down Dockerhub acceptance of images, and we aren't going to provide a templated Dockerfile because none of us care for the ones we've seen. We might change our minds on the latter when we've had a chance to see how much our Dockerfiles change in the future and how much updating the doc needs as a result. I've added comments to the Makefiles and Dockerfiles to update the doc when those files change. I've also added a paragraph describing how to use the repo more or less as-is, changing only the name of the base image.